MIGRATING-0.6-0.7.md

```diff
@@ -396,4 +288,259 @@ Note that the table above shows the new fonts' `ascii` variants only. Some new f
+
+```rust
+use embedded_graphics::{
+    draw_target::{DrawTarget, DrawTargetExt},
+    geometry::{OriginDimensions, Size},
+    image::ImageDrawable,
+    pixelcolor::{PixelColor, Rgb888},
+    primitives::Rectangle,
+};
+
+struct MyRgb888Image {
+  // ...
+}
+
+impl ImageDrawable for MyRgb888Image {
+    type Color = Rgb888;
+
+    fn draw<D>(&self, target: &mut D) -> Result<(), D::Error>
+    where
+        D: DrawTarget<Color = Rgb888>,
+    {
+        // Draw the image to the target, e.g. by calling `target.fill_contiguous` or by using another `Drawable`.
+    }
+
+    fn draw_sub_image<D>(&self, target: &mut D, area: &Rectangle) -> Result<(), D::Error>
+    where
+        D: DrawTarget<Color = Self::Color>,
+    {
+        // Delegate to the draw() method using a reduced draw target
+        self.draw(&mut target.translated(-area.top_left).clipped(area))
+    }
+}
+
+impl OriginDimensions for MyRgb888Image {
+    fn size(&self) -> Size {
+        // Return image width and height in pixels
+    }
+}
+```
+
+## For text rendering crates
+
+### Monospace fonts
+
+Monospaced fonts should now be built by creatign a `MonoFont` struct and assigning it to a `const`. The following example shows a migration of a font with 5x9 pixel characters using the Latin-1 encoding.
+
+```diff
+// The font bitmap has 32 character glyphs per row.
+const CHARS_PER_ROW: u32 = 32;
+
+// Each character is 5x9 px.
+const GLYPH_SIZE: Size = Size::new(5, 9);
+
+- // Map a given character to an index in the glyph bitmap
+- fn char_offset_impl(c: char) -> u32 {
+-     let fallback = '?' as u32 - ' ' as u32;
+-     if c < ' ' {
+-         return fallback;
+-     }
+-     if c <= '~' {
+-         return c as u32 - ' ' as u32;
+-     }
+-     if c < '\u{00A0}' || c > 'ÿ' {
+-         return fallback;
+-     }
+-     c as u32 - ' ' as u32 - 33
+- }
+-
+- #[derive(Debug, Copy, Clone)]
+- pub struct ExampleFont {}
+- impl Font for ExampleFont {
+-     const FONT_IMAGE: &'static [u8] = include_bytes!("../data/ExampleFont.raw");
+-     const CHARACTER_SIZE: Size = Size::new(5, 9);
+-     const FONT_IMAGE_WIDTH: u32 = Self::CHARACTER_SIZE.width * CHARS_PER_ROW;
+-
+-     fn char_offset(c: char) -> u32 {
+-         char_offset_impl(c)
+-     }
+- }
++ use embedded_graphics::{
++     geometry::Size,
++     image::ImageRaw,
++     mono_font::{mapping::StrGlyphMapping, DecorationDimensions, MonoFont},
++ };
++
++ pub const EXAMPLE_FONT: MonoFont = MonoFont {
++     image: ImageRaw::new_binary(
++         // In this example, this equals 32 characters per row, each character 5px across.
++         160,
++     ),
++     // A glyph mapping with base ASCII range and Latin1 range.
++     glyph_mapping: &StrGlyphMapping::new("\0 ~\0\u{00A0}ÿ", '?' as usize - ' ' as usize),
++     character_size: Size::new(5, 9),
++     character_spacing: 0,
++     baseline: 7,
++     underline: DecorationDimensions::new(8, 1),
++     strikethrough: DecorationDimensions::new(4, 1),
++ };
```

Comment:
I would suggest to tweak the example a bit, so that we can use a predefined mapping:

```suggestion
- // Map a given character to an index in the glyph bitmap
- fn char_offset_impl(c: char) -> u32 {
-     let fallback = '?' as u32 - ' ' as u32;
-     if c < ' ' {
-         return fallback;
-     }
-     if c <= '\u{007f}' {
-         return c as u32 - ' ' as u32;
-     }
-     if c < '\u{00A0}' || c > 'ÿ' {
-         return fallback;
-     }
-     c as u32 - ' ' as u32 - 32
- }
-
- #[derive(Debug, Copy, Clone)]
- pub struct ExampleFont {}
- impl Font for ExampleFont {
-     const FONT_IMAGE: &'static [u8] = include_bytes!("../data/ExampleFont.raw");
-     const CHARACTER_SIZE: Size = Size::new(5, 9);
-     const FONT_IMAGE_WIDTH: u32 = Self::CHARACTER_SIZE.width * CHARS_PER_ROW;
-
-     fn char_offset(c: char) -> u32 {
-         char_offset_impl(c)
-     }
- }
+ use embedded_graphics::{
+     geometry::Size,
+     image::ImageRaw,
+     mono_font::{mapping::ISO_8859_1, DecorationDimensions, MonoFont},
+ };
+
+ pub const EXAMPLE_FONT: MonoFont = MonoFont {
+     image: ImageRaw::new_binary(
+         // This example uses 32 characters per row, each character 5px across.
+         32 * 5,
+     ),
+     glyph_mapping: &ISO_8859_1,
+     character_size: Size::new(5, 9),
+     character_spacing: 0,
+     baseline: 7,
+     underline: DecorationDimensions::new(8, 1),
+     strikethrough: DecorationDimensions::new(4, 1),
+ };
```